### ISSUE TITLE
azuread_application access_token_issuance_enabled churns when set to default null value

### ISSUE THREAD

**Issue by dekimsey** [hashicorp/terraform-provider-azuread#483]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureAD Provider) Version

```
$ terraform -v
Terraform v1.0.0
on darwin_amd64
+ provider registry.terraform.io/hashicorp/azuread v1.6.0
```

### Affected Resource(s)

* `azuread_application`

### Terraform Configuration Files

```hcl
resource "azuread_application" "this" {
  display_name = "my-app"
  web {
    homepage_url = var.homepage
    redirect_uris =  ["${var.url}/oauth2/idpresponse"]
    # If set to null (or not provided), resource will churn
    #implicit_grant {
    #  access_token_issuance_enabled = false
    #}
  }
  identifier_uris = [var.url]
}
```

### Expected Behavior

Resource should not churn.

### Actual Behavior

```
$ terraform apply
Terraform will perform the following actions:

  # module.azure-app.azuread_application.this will be updated in-place
  ~ resource "azuread_application" "this" {
        id                             = "dead-beef"
        name                           = "Tracing"
        # (16 unchanged attributes hidden)


      ~ web {
            # (2 unchanged attributes hidden)

          - implicit_grant {
              - access_token_issuance_enabled = false -> null
            }
        }
        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### Steps to Reproduce

1. `terraform apply`
2. `terraform apply`

### Important Factoids

None

### References

None

**Comment by manicminer** [comment]
Hi @dekimsey, thanks for reporting this bug!

Due to the particular schema configuration for this property, the only viable fix without breaking existing configurations is to make the `implicit_grant` block Computed. This will prevent the persistent diff you are seeing, however it has the side effect of making Terraform ignore the change when you remove it from your configuration. For example, if you had set `access_token_issuance_enabled = true` and then remove this line, Terraform will not consider this changed and removing it will have no effect - meaning you'll have to explicitly set this to false to unset it. Doing just that means you won't encounter the persistent diff in the first place!

In short, I'm not certain this is worth 'fixing' - there is no true fix without releasing a breaking schema change. We've done exactly that in the upcoming v2.0, without making these properties Computed, so you'll enjoy the correct behavior all round in v2.0. As such, and since the v1 lineage has now been archived except for major bugs, I'm going to close this issue. As mentioned, this won't happen in v2.0. Thanks!
